### DISCUSSION TITLE
obspy functionality holes

### DISCUSSION THREAD

**Discussion by pavlis** [mspass-team/mspass#140] [Design & Development]
I have run across what is a somewhat absurd hole in obspy's functionalty.    It has been raised, I think, on their github site but does not seem to have been corrected.  The crazy issue is that they have an excellent travel time calculator for global earth models using the standard, fast taup method.  They do not, however, seem to have a simple distance calculator to compute great circle path distances.  They recommend a web service call to the iris client that apparently supports a distaz calculation.  There is, no doubt, a stock python package out there to do that calculation, but I think it would only be a few hours work to convert a fairly substantial C library I have from the open source version of antelope that was called "Datascope".  The library that had was called coords and has some of the following:
dist - t return an epicentral distance and azimuth between two points
latlon - kind of the opposite of dist.  Given one point, a distance and azimuth, return the coordinates of the end of the great circle path.  

Maybe we should start our own coords module?

**Comment by wangyinz** [comment]
Interesting. Yep, I don't see a reason not to.

**Reply by wangyinz** [reply]
btw, could you please post the link to the ObsPy issue that you referred to? It is important for bookkeeping.

**Reply by pavlis** [reply]
I appear to have been incorrect.  While searching for the pages I saw in issues a google search with a different incantation revealed [this page](https://docs.obspy.org/packages/autogen/obspy.geodetics.base.gps2dist_azimuth.html#obspy.geodetics.base.gps2dist_azimuth).  It does exactly what is needed with some even fancier options with ellipticity.   I will use that function in the getting_started tutorial draft.
